### PR TITLE
feat: add `array/base/broadcasted-quinary4d`

### LINKED ISSUE #3171: [RFC]: add `array/base/broadcasted-quinary4d`
### Description

This RFC proposes adding a new package to the `array/base/*` namespace: `@stdlib/array/base/broadcasted-quinary4d`.

This package is the quinary equivalent of [`@stdlib/array/base/broadcasted-binary4d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/broadcasted-binary4d).

To provide a concrete example of what a PR adding the desired package should contain, see [`@stdlib/array/base/broadcasted-quinary2d`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/array/base/broadcasted-quinary2d), which is the 2D package equivalent.

While the package proposed in this RFC will **not** match the implementations found in the above packages, those packages should provide a conceptual idea of what is desired. Do **not** simply copy-paste the code found in those packages.

### Notes

- If you are interested in contributing a PR which addresses this RFC and are a **first-time contributor** or not yet familiar with our project conventions, please do **not** submit LLM-generated code. Please consult our [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md) and the associated [development guide](https://github.com/stdlib-js/stdlib/blob/develop/docs/development.md). Failure to respect project conventions will result in your PR being rejected without review. Thank you for understanding!

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `[RFC]:`.

### PR DESCRIPTION
**Pull request by Neerajpathak07**
Resolves #3171 

## Description

> What is the purpose of this pull request?

This pull request:

-   adds implementation for `array/base/broadcasted-quinary4d`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3171 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-  [x] Read, understood, and followed the [contributing guidelines](https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md)

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+		k2 = 0;
+	    m2 = 0;
+	    n2 = 0;
+	    p2 = 0;
+	    x2 = x[ j3 ];
+	    y2 = y[ k3 ];
+	    z2 = z[ m3 ];
+	    w2 = w[ n3 ];
+	    u2 = u[ p3 ];
+	    v2 = v[ i3 ];
```
Comment (AbhijitRaut04) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js]:
```suggestion
		k2 = 0;
	        m2 = 0;
	        n2 = 0;
	        p2 = 0;
	        x2 = x[ j3 ];
	        y2 = y[ k3 ];
	        z2 = z[ m3 ];
	        w2 = w[ n3 ];
	        u2 = u[ p3 ];
	        v2 = v[ i3 ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	        k1 = 0;
+	        m1 = 0;
+	        n1 = 0;
+	        p1 = 0;
+	        x1 = x2[ j2 ];
+	        y1 = y2[ k2 ];
+	        z1 = z2[ m2 ];
+	        w1 = w2[ n2 ];
+	        u1 = u2[ p2 ];
+	        v1 = v2[ i2 ];
```
Comment (AbhijitRaut04) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js]:
```suggestion
			j1 = 0;
	                k1 = 0;
	                m1 = 0;
	                n1 = 0;
	                p1 = 0;
	                x1 = x2[ j2 ];
	                y1 = y2[ k2 ];
	                z1 = z2[ m2 ];
	                w1 = w2[ n2 ];
	                u1 = u2[ p2 ];
	                v1 = v2[ i2 ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+	    	    y0 = y1[ k1 ];
+	    	    z0 = z1[ m1 ];
+	    	    w0 = w1[ n1 ];
+	    	    u0 = u1[ p1 ];
+	    	    v0 = v1[ i1 ];
+		        for ( i0 = 0; i0 < S0; i0++ ) {
+			        v0[ i0 ] = fcn( x0[ j0 ], y0[ k0 ], z0[ m0 ], w0[ n0 ], u0[ p0 ] ); // eslint-disable-line max-len
+			        j0 += dx0;
+			        k0 += dy0;
+					m0 += dz0;
```
Comment (AbhijitRaut04) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js]:
```suggestion
	    	            j0 = 0;
	    	            k0 = 0;
	    	            m0 = 0;
	    	            n0 = 0;
	    	            p0 = 0;
	    	            x0 = x1[ j1 ];
	    	            y0 = y1[ k1 ];
	    	            z0 = z1[ m1 ];
	    	            w0 = w1[ n1 ];
	    	            u0 = u1[ p1 ];
	    	            v0 = v1[ i1 ];
		                for ( i0 = 0; i0 < S0; i0++ ) {
			                v0[ i0 ] = fcn( x0[ j0 ], y0[ k0 ], z0[ m0 ], w0[ n0 ], u0[ p0 ] ); // eslint-disable-line max-len
			                j0 += dx0;
			                k0 += dy0;
					m0 += dz0;
```

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+			        n0 += dw0;
+			        p0 += du0;
+		        }
+		        j1 += dx1;
+		        k1 += dy1;
+		        m1 += dz1;
+		        n1 += dw1;
+		        p1 += du1;
+	        }
+			j2 += dx2;
```
Comment (AbhijitRaut04) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js]:
```suggestion
			                n0 += dw0;
			                p0 += du0;
		                }
		                j1 += dx1;
		                k1 += dy1;
		                m1 += dz1;
		                n1 += dw1;
		                p1 += du1;
	                }
			j2 += dx2;
```

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js (hunk 1 above)
Comment (Neerajpathak07) [reply lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/lib/main.js]: @AbhijitRaut04 I have looked upon the changes and corrected the error.
Thanks for the help!!

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    Examples
+    --------
+    > function fcn( x, y, z, w, v ) { return x + y + z + w + v; };
+    > var x = [ 1.0, 2.0 ];
+    > var y = [ [ [ [ 3.0 ], [ 4.0 ] ] ] ];
+    > var z = [ [ [ [ 1.0 ] ] ] ];
+    > var w = [ 2.0 ];
+    > var v = [ 1.0 ];
+    > var out = [ [ [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ] ] ];
+    > var shapes = [ [ [ [ 2 ], [ 2, 1 ], [ 1, 1 ], [ 1 ], [ 1 ], [ 2, 2 ] ] ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt]: @Neerajpathak07 This array is too deeply nested. 
```suggestion
    > var shapes = [ [ 2 ], [ 2, 1 ], [ 1, 1 ], [ 1 ], [ 1 ], [ 2, 2 ] ];
```

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+function add( x, y, z, w, v ) {
+    return x + y + z + w + v;
+}
+
+var x = [ [ [ [ 1.0, 2.0 ] ] ] ];
+var y = [ [ [ [ 3.0 ], [ 4.0 ] ] ] ];
+var z = [ [ [ [ 5.0 ] ] ] ];
+var w = [ [ [ [ 2.0 ] ] ] ];
+var v = [ [ [ [ 1.0 ] ] ] ];
+var out = zeros4d( [ 2, 2 ] );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md]: You are using `zeros4d`, but are only provided a 2-dimensional shape. You need to provide all four dimensions.

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt (hunk 5 above)
Comment (Neerajpathak07) [reply lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt]: I made the suggested changes an I'll push them soon

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md (hunk 6 above)
Comment (Neerajpathak07) [reply lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md]: I tried changing the dimensions for `zeros4d` to [ 2, 2, 2, 2 ] but its still throwing error.

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+var w = [ [ [ [ 2.0 ] ] ] ];
+var v = [ [ [ [ 1.0 ] ] ] ];
+var out = zeros4d( [ 2, 2, 2, 2 ] );
+
+var shapes = [
+    [ 1, 1, 1, 2 ],
+    [ 1, 1, 2, 1 ],
+    [ 1, 1, 2, 2 ],
+    [ 1, 2, 1, 1 ],
+    [ 2, 2, 2, 2 ],
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/README.md]: @Neerajpathak07 These shapes are incorrect. For `z`, `w`, and `v` the shape is `[ 1, 1, 1, 1 ]`.

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+    Examples
+    --------
+    > function fcn( x, y, z, w, v ) { return x + y + z + w + v; };
+    > var x = [ 1.0, 2.0 ];
+    > var y = [ [ [ [ 3.0 ], [ 4.0 ] ] ] ];
+    > var z = [ [ [ [ 1.0 ] ] ] ];
+    > var w = [ 2.0 ];
+    > var v = [ 1.0 ];
+    > var out = [ [ [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ] ] ];
+    > var shapes = [ [ 2 ], [ 2, 1 ], [ 1, 1 ], [ 1 ], [ 1 ], [ 2, 2 ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt]: @Neerajpathak07 These shapes are incorrect; hence, the lint errors.

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/package.json [hunk 9]
```diff
@@ -0,0 +1,67 @@
+{
+    "name": "@stdlib/array/base/broadcasted-quinary4d",
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/package.json]: This file has incorrect indentation. Should be 2-space, hence the CI lint error.

File: lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    Examples
+    --------
+    > function fcn( x, y, z, w, v ) { return x + y + z + w + v; };
+    > var x = [ [ [ [ 1.0, 2.0 ] ] ] ];
+    > var y = [ [ [ [ 3.0 ], [ 4.0 ] ] ] ];
+    > var z = [ [ [ [ 1.0 ] ] ] ];
+    > var w = [ [ [ [ 2.0 ] ] ] ];
+    > var v = [ [ [ [ 1.0 ] ] ] ];
+    > var out = [ [ [ [ 0.0, 0.0 ], [ 0.0, 0.0 ] ] ] ];
+    > var shapes = [ [ 1, 1, 1, 2 ], [ 1, 1, 2, 1 ], [ 1, 1, 1, 1 ], [ 1, 1, 1, 1 ], [ 1, 1, 1, 1 ], [ 2, 2, 2, 2 ] ];
```
Comment (kgryte) [inline lib/node_modules/@stdlib/array/base/broadcasted-quinary4d/docs/repl.txt]:
```suggestion
    > var shapes = [ [ 1, 1, 1, 2 ], [ 1, 1, 2, 1 ], [ 1, 1, 1, 1 ], [ 1, 1, 1, 1 ], [ 1, 1, 1, 1 ], [ 1, 1, 2, 2 ] ];
```

The output array shape is incorrect; hence, the lint errors.

**Comment by Neerajpathak07** [comment]
@kgryte all CI checks are passing!

**Review by kgryte — APPROVED** [review]

**Pull request merged**